spring-cloud-gcp-autoconfigure/src/test/java/com/google/cloud/spring/autoconfigure/pubsub/health/PubSubHealthIndicatorAutoConfigurationTests.java

```diff
@@ -46,37 +64,131 @@
 			.withBean(GcpProjectIdProvider.class,  () -> () -> "fake project")
 			.withBean(CredentialsProvider.class, () -> () -> mock(Credentials.class));
 
+	@SuppressWarnings("unchecked")
 	@Test
-	public void healthIndicatorPresent() {
+	public void healthIndicatorPresent() throws Exception {
+		PubSubTemplate mockPubSubTemplate = mock(PubSubTemplate.class);
+		ListenableFuture<List<AcknowledgeablePubsubMessage>> future = mock(ListenableFuture.class);
+
+		when(future.get(anyLong(), any())).thenReturn(Collections.emptyList());
+		when(mockPubSubTemplate.pullAsync(anyString(), anyInt(), anyBoolean())).thenReturn(future);
+
 		this.baseContextRunner
-				.withPropertyValues("management.health.pubsub.enabled=true")
+				.withBean("pubSubTemplate", PubSubTemplate.class, () -> mockPubSubTemplate)
+				.withPropertyValues(
+						"management.health.pubsub.enabled=true",
+						"spring.cloud.gcp.pubsub.health.subscription=test",
+						"spring.cloud.gcp.pubsub.health.timeout-millis=1500",
+						"spring.cloud.gcp.pubsub.health.acknowledgeMessages=true")
 				.run(ctx -> {
 					PubSubHealthIndicator healthIndicator = ctx.getBean(PubSubHealthIndicator.class);
 					assertThat(healthIndicator).isNotNull();
```

Comment:
Since this is testing context loading, might want to verify that the health indicator validation got invoked (that template.pullAsync() got called)